src/utils/generateChartData.js

```diff
@@ -0,0 +1,62 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+import { metricToChartName } from "../constants/metrics";
+
+const generateChartData = (data, metrics, period = 60) => {
+  const datasets = metrics.map((metric) => ({
+    metric,
+    label: metricToChartName[metric],
+    data: [],
+  }));
+  const labels = [];
+
+  const { latestYear, latestMonth } = metrics.reduce(
+    (acc, metric) => {
+      const { year, month } = data[metric][data[metric].length - 1];
+
+      if (year > acc.latestYear) return { latestYear: year, latestMonth: month };
+      if (month > acc.latestMonth) return { latestYear: year, latestMonth: month };
```

Comment:
Does this work as expected? If we see a data point with year=2016, month=5 and then later we see a datapoint with year=2014, month=8, wouldn't see return latestYear=2014, latestMonth=8?

Reply:
@jessex My fault, thanks!